hs_core/models.py

```diff
@@ -1592,23 +1623,27 @@ def get_xml(self, pretty_print=True):
             hsterms_pub_url = etree.SubElement(dc_pub_rdf_Description, '{%s}publisherURL' % self.NAMESPACES['hsterms'])
             hsterms_pub_url.set('{%s}resource' % self.NAMESPACES['rdf'], self.publisher.url)
 
+        res_url_root = '{site_url}/resource/'.format(site_url = current_site_url())
         for rel in self.relations.all():
             dc_relation = etree.SubElement(rdf_Description, '{%s}relation' % self.NAMESPACES['dc'])
             dc_rel_rdf_Description = etree.SubElement(dc_relation, '{%s}Description' % self.NAMESPACES['rdf'])
-            rel_dcterm = '{%s}' + rel.type
-            dcterms_type = etree.SubElement(dc_rel_rdf_Description, rel_dcterm % self.NAMESPACES['dcterms'])
-            # check if the relation value starts with 'http://' or 'https://'
-            if rel.value.lower().find('http://') == 0 or rel.value.lower().find('https://') == 0:
-                dcterms_type.set('{%s}resource' % self.NAMESPACES['rdf'], rel.value)
+            if rel.type in Relation.HS_RELATION_TERMS:
+                term_ns = self.NAMESPACES['hsterms']
             else:
-                dcterms_type.text = rel.value
+                term_ns = self.NAMESPACES['dcterms']
+            terms_type = etree.SubElement(dc_rel_rdf_Description, '{%s}%s' % (term_ns, rel.type))
+
+            if rel.value.lower().find(res_url_root) == 0:
```

Comment:
@horsburgh @pkdash After giving this a bit of more thought, I feel the new code makes more sense. For example, for the newly added "Copied From" relation, the relation value is a URL that points to a host site where the current HydroShare resource is copied from. With the old code logic, this host site URL will be encoded using rdf:resource, and with the new code, this host url will be encoded directly with the url value, and only hydroshare resource will be encoded using rdf:resource. @horsburgh I will wait for your input regarding this, then either keep the new code, or change it back to the old code accordingly.